pkg/segment/structs/querystructs.go

```diff
@@ -35,11 +35,12 @@ import (
 
 // New struct for passin query params
 type QueryContext struct {
-	TableInfo *TableInfo
-	SizeLimit uint64
-	Scroll    int
-	Orgid     uint64
-	RawQuery  string
+	TableInfo   *TableInfo
+	SizeLimit   uint64
+	Scroll      int
+	Orgid       uint64
+	RawQuery    string
+	SearchTerms []string
```

Comment:
So for `foo=bar AND baz=42` the `SearchTerms` would be `bar` and `42`, right? What about `foo!=bar` and `baz>42`? Are we not handling those cases? What about `NOT foo=bar`?